packages/webapp/src/containers/Animals/Inventory/useAnimalOrBatchRemoval.ts

```diff
@@ -0,0 +1,143 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { Dispatch, SetStateAction, useState, useEffect } from 'react';
+import {
+  useDeleteAnimalBatchesMutation,
+  useDeleteAnimalsMutation,
+  useRemoveAnimalBatchesMutation,
+  useRemoveAnimalsMutation,
+} from '../../../store/api/apiSlice';
+import { toLocalISOString } from '../../../util/moment';
+import { parseInventoryId } from '../../../util/animal';
+import { CREATED_IN_ERROR_ID, FormFields } from '../../../components/Animals/RemoveAnimalsModal';
+import useMutations from '../../../hooks/api/useMutations';
+import { enqueueErrorSnackbar, enqueueSuccessSnackbar } from '../../Snackbar/snackbarSlice';
+import { TFunction } from 'i18next';
+
+const useAnimalOrBatchRemoval = (
+  selectedInventoryIds: string[],
+  setSelectedInventoryIds: Dispatch<SetStateAction<string[]>>,
+  dispatch: Dispatch<any>,
+  t: TFunction,
+) => {
+  const [removeAnimals] = useRemoveAnimalsMutation();
+  const [removeBatches] = useRemoveAnimalBatchesMutation();
+
+  const { mutations, isLoading, isError, isSuccess } = useMutations([
+    { label: 'deleteAnimals', hook: useDeleteAnimalsMutation },
+    { label: 'deleteBatches', hook: useDeleteAnimalBatchesMutation },
+  ]);
+
+  const [removalModalOpen, setRemovalModalOpen] = useState(false);
+
+  const removeFromSelectedIds = (removeTheseIds: string[]) => {
+    const newArray = selectedInventoryIds;
+    removeTheseIds.forEach((removalId: string): void => {
+      const index = newArray.indexOf(removalId);
+      if (index > -1) {
+        // only splice array when item is found
+        newArray.splice(index, 1); // 2nd parameter means remove one item only
+      }
+    });
+    return newArray;
+  };
```

Comment:
Could you please use filter instead of splice loop

```suggestion
  const removeFromSelectedIds = (removeTheseIds: string[]) => {
    return selectedInventoryIds.filter((id) => !removeTheseIds.includes(id));
  };
```

Reply:
Oh yeah thats better haha 😂  .. removed this function and converted to a one liner so no more goofy names either like removeTheseIds haha.